File: OOP_Variable/4.py
class C(object):
    def __init__(self, v1):
        self.__value = v1   ## __value: 인스턴스 밖에서 접근 불가

    def show(self):
        print(self.__value)

#파이썬: 인스턴스변수에 접근이 허용됨
# __가 붙어있으면 외부에서 접근 금지됨
c1 = C(10)
# print(c1.__value)
c1.show()   # __변수도 접근 가능


# print(c1.value)
# c1.value = 20
# print(c1.value)
# c1.show()

File: OOP_Variable/4.rb
class C
  # attr_reader :value  # value라는 속성을 읽기 가능하도록 함
  # attr_writer :value  # 쓰기 가능하도록 함
  attr_accessor :value  #읽기쓰기 가능하도록 함
  #생성자
  def initialize(v1)
    @value = v1 #내부
  end
  def show()
    p @value  #내부
  end
end
#루비: 인스턴스변수에 접근이 허용되지 않음
c1 = C.new(10)
p c1.value  #외부, 속성: 읽기 가능

c1.value = 20 #   attr_writer :value
p c1.value
